add categoryTaskScreen

diyar214 opened the pull request:
### **What I have did:**

- I added the `Category Task Screen`.
- I added Placeholder for view model and UI state.

### **What is left**
- The status bar color:  I tried to change color using `systemuicontroller`, but I have failed, and I used other ways, and they also failed.

### **Screenshots**

<img width="153" alt="Screenshot 2025-06-18 at 22 53 12" src="https://github.com/user-attachments/assets/f79a2a0e-b847-4963-beab-3888bbf22411" />
<img width="153" alt="Screenshot 2025-06-18 at 22 55 29" src="https://github.com/user-attachments/assets/7e58094b-522c-462c-bc6d-64b76e8bffcd" />
<img width="153" alt="Screenshot 2025-06-18 at 22 55 40" src="https://github.com/user-attachments/assets/9897f99c-c9a9-47b9-9732-06604420cbf8" />

<img width="153" alt="Screenshot 2025-06-18 at 23 14 39" src="https://github.com/user-attachments/assets/6a038489-3f93-4f53-af33-dce7fd35588f" />
<img width="153" alt="Screenshot 2025-06-18 at 23 14 51" src="https://github.com/user-attachments/assets/59e2bcca-0119-4f59-be5c-2de59a38079c" />
<img width="153" alt="Screenshot 2025-06-18 at 23 14 59" src="https://github.com/user-attachments/assets/a4a86cb3-83a1-41c3-8bd6-4c1b9e8ee272" />

EL-MANCY committed:
add coil dependency
add AddNewCategorey.kt screen design
add UploadBox.kt as a component

EL-MANCY committed:
add a drop shadow to the buttons container
remove fillMaxHeight and wrapped the content with right dimensions

EL-MANCY committed:
-fix padding of the feature/addNewCategoreyBottomSheet

-fix the stroke of UploadBox.kt

EL-MANCY committed:
process the selected image to bitmap

diyar214 committed:
add categoryTaskScreen

diyar214 committed:
add placeholder for view model and ui state

haideraljawasim commented:
Good job,
small changes needed 
Last thing you didn't add edit button logic or view !!

Ahmed7osny1 committed:
add category screen and category view model

Ahmed7osny1 committed:
Merge remote-tracking branch 'origin/dev' into feature/category

# Conflicts:
#	app/build.gradle.kts

Ahmed7osny1 committed:
update category view model

EL-MANCY committed:
-add HelperFunctions.kt for the processImage
-add some string resources
-extract multiple parts of the composables to small functions

EL-MANCY committed:
Merge remote-tracking branch 'refs/remotes/origin/dev' into feature/addNewCategoreyBottomSheet

# Conflicts:
#	app/build.gradle.kts
#	app/src/main/res/values/strings.xml

EL-MANCY committed:
hoist multiple states

EL-MANCY committed:
Merge remote-tracking branch 'origin/dev' into feature/category

# Conflicts:
#	app/src/main/java/com/sanaa/tudee_assistant/TudeeAssistantApplication.kt

EL-MANCY committed:
Merge remote-tracking branch 'origin/feature/addNewCategoreyBottomSheet' into feature/category

EL-MANCY committed:
use the viewmodel and attach the bottom sheet to CategoryScreen.kt

diyar214 committed:
Merge branch 'dev' into category-tasks-screen

diyar214 committed:
change screen path

diyar214 committed:
Merge branch 'feature/category' into category-tasks-screen

AimanYosofiUlt commented:
> ### **What I have did:**
> * I added the `Category Task Screen`.
> * I added Placeholder for view model and UI state.
> 
> ### **What is left**
> * The status bar color:  I tried to change color using `systemuicontroller`, but I have failed, and I used other ways, and they also failed.
> 
> ### **Screenshots**
> <img alt="Screenshot 2025-06-18 at 22 53 12" width="153" src="https://private-user-images.githubusercontent.com/101934484/456690291-f79a2a0e-b847-4963-beab-3888bbf22411.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTAzNjIyNTAsIm5iZiI6MTc1MDM2MTk1MCwicGF0aCI6Ii8xMDE5MzQ0ODQvNDU2NjkwMjkxLWY3OWEyYTBlLWI4NDctNDk2My1iZWFiLTM4ODhiYmYyMjQxMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNjE5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDYxOVQxOTM5MTBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iYzFjN2YwYTVjZWQ3MzI1NTllNmM1Nzc1MzI1ZDNkODFlODMwM2ZhNWI3M2U4Y2M2NmI2ZWIyYjIxYWI2ODI3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.YVYKZ5dbrgizpZuz1TRN8uvHNTFFAD5a4dayCjCx31s"> <img alt="Screenshot 2025-06-18 at 22 55 29" width="153" src="https://private-user-images.githubusercontent.com/101934484/456690294-7e58094b-522c-462c-bc6d-64b76e8bffcd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTAzNjIyNTAsIm5iZiI6MTc1MDM2MTk1MCwicGF0aCI6Ii8xMDE5MzQ0ODQvNDU2NjkwMjk0LTdlNTgwOTRiLTUyMmMtNDYyYy1iYzZkLTY0Yjc2ZThiZmZjZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNjE5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDYxOVQxOTM5MTBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lMDc1MTYyODYyN2VmZjBjMmM3ZTFiZmFmYWFlYjVkMGI1MDBkOGQwYTlhMzMxYWRmM2I5OTZiOGQ3MzU5MTdjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.hv-UV5WPgoYyia4BLHTeqbYwcdk4n5rEyBgVop2acSo"> <img alt="Screenshot 2025-06-18 at 22 55 40" width="153" src="https://private-user-images.githubusercontent.com/101934484/456690281-9897f99c-c9a9-47b9-9732-06604420cbf8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTAzNjIyNTAsIm5iZiI6MTc1MDM2MTk1MCwicGF0aCI6Ii8xMDE5MzQ0ODQvNDU2NjkwMjgxLTk4OTdmOTljLWM5YTktNDdiOS05NzMyLTA2NjA0NDIwY2JmOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNjE5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDYxOVQxOTM5MTBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xZDhiMGZjZjZlOWQzMThlNGJhMGVjYWY5MzZlYmU3ZGEzYmJiOGI4ODgyZjNiMjM4NmFiYTAxMzkxODM1ZThiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.RCRDPoPGmyvVcZ0ZVniuhe4dHCPxR9Emvte8J0bDZy0"> <img alt="Screenshot 2025-06-18 at 23 14 39" width="153" src="https://private-user-images.githubusercontent.com/101934484/456697810-6a038489-3f93-4f53-af33-dce7fd35588f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTAzNjIyNTAsIm5iZiI6MTc1MDM2MTk1MCwicGF0aCI6Ii8xMDE5MzQ0ODQvNDU2Njk3ODEwLTZhMDM4NDg5LTNmOTMtNGY1My1hZjMzLWRjZTdmZDM1NTg4Zi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNjE5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDYxOVQxOTM5MTBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jNTkwMTQzMjg1YWQ1ZGM4Yzg1ZjU3MzE3YzA1N2EzOTZiY2YyYWRlYzUyZTJkNzU3M2IwNjVkNTA0OTVhMDY2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.O38ywjF0-eqaVY8vnBiDmY6sdeCXWZJ9Pzdk_aypg7o"> <img alt="Screenshot 2025-06-18 at 23 14 51" width="153" src="https://private-user-images.githubusercontent.com/101934484/456697807-59e2bcca-0119-4f59-be5c-2de59a38079c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTAzNjIyNTAsIm5iZiI6MTc1MDM2MTk1MCwicGF0aCI6Ii8xMDE5MzQ0ODQvNDU2Njk3ODA3LTU5ZTJiY2NhLTAxMTktNGY1OS1iZTVjLTJkZTU5YTM4MDc5Yy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNjE5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDYxOVQxOTM5MTBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kNDA0ZTgxYzE0NWU2NzI4NjgzODdmMjQwMWJjZDY1NGI2ODlkN2E3YTA2ZGE2MjA0ZjIyMjQwZWNjMDJjZmMwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.8wVu5MfOlxYR7Q11Uq4qow_U-aZHarTEPQi5DQdrr6s"> <img alt="Screenshot 2025-06-18 at 23 14 59" width="153" src="https://private-user-images.githubusercontent.com/101934484/456697812-a4a86cb3-83a1-41c3-8bd6-4c1b9e8ee272.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTAzNjIyNTAsIm5iZiI6MTc1MDM2MTk1MCwicGF0aCI6Ii8xMDE5MzQ0ODQvNDU2Njk3ODEyLWE0YTg2Y2IzLTgzYTEtNDFjMy04YmQ2LTRjMWI5ZThlZTI3Mi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNjE5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDYxOVQxOTM5MTBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wNTExOThhOWY4MGJlMGRmYTY1MTUyZGI1YjZiZjhjMjUxZDU0Y2EwYjljNzM4ZWIxZDA5MDJjODBjZGRhNDNmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.fUxWE8JbG46plLgPTZvanqZK64hpHZo73CRhTdtt9xU">

You add the branch as main branch, it should be inside `feature/...` folder, just as note for next time

AimanYosofiUlt reviewed (changes requested):
Good Job, just see me leatest comments
